feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.ts

```diff
@@ -19,51 +28,200 @@ import { ConfiguratorAttributeBaseComponent } from '../base/configurator-attribu
 export class ConfiguratorAttributeCheckBoxListComponent
   extends ConfiguratorAttributeBaseComponent
   implements OnInit {
+  attributeCheckBoxForms = new Array<FormControl>();
+  loading$ = new BehaviorSubject<boolean>(false);
+
   @Input() attribute: Configurator.Attribute;
   @Input() group: string;
   @Input() ownerKey: string;
 
   @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
 
+  // TODO(#11681): make config a required dependency
+  /**
+   * default constructor
+   * @param {ConfiguratorAttributeQuantityService} quantityService
+   */
+  constructor(
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    configUtilsService: ConfiguratorStorefrontUtilsService,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    quantityService: ConfiguratorAttributeQuantityService
+  );
+
+  /**
+   * @deprecated since 3.3
+   */
+  constructor(configUtilsService: ConfiguratorStorefrontUtilsService);
+
   constructor(
-    protected configUtilsService: ConfiguratorStorefrontUtilsService
+    protected configUtilsService: ConfiguratorStorefrontUtilsService,
+    protected quantityService?: ConfiguratorAttributeQuantityService
   ) {
     super();
   }
 
-  attributeCheckBoxForms = new Array<FormControl>();
-
   ngOnInit() {
+    const selectedValues = this.attribute.values.filter(
+      (value) => value.selected
+    );
+
     for (const value of this.attribute.values) {
       let attributeCheckBoxForm;
+
       if (value.selected === true) {
-        attributeCheckBoxForm = new FormControl(true);
+        attributeCheckBoxForm = new FormControl({
+          value: true,
+          disabled:
+            this.attribute.required && selectedValues.length < 2 ? true : false,
```

Comment:
here can be this: `disabled: !this.allowZeroValueQuantity`, right?

Reply:
pick